src/Type/Trout/Client.purs

```diff
@@ -27,17 +33,38 @@ import Type.Trout.ContentType.JSON (JSON)
 import Type.Trout.PathPiece (class ToPathPiece, toPathPiece)
 import Type.Trout.Record as Record
 
+newtype BaseURI = BaseURI { scheme :: URIScheme , authority :: Authority }
+
+derive instance eqBaseURI :: Eq BaseURI
+derive instance genericBaseURI :: Generic BaseURI _
+derive instance newtypeBaseURI :: Newtype BaseURI _
+derive instance ordBaseURI :: Ord BaseURI
+instance showScheme :: Show BaseURI where
+  show = genericShow
+
+instance hasClientsBaseURI :: ( HasClients sub subMk
+                              )
+                              => HasClients (BaseURI :> sub) (BaseURI -> subMk) where
+  getClients _ req (BaseURI {authority, scheme}) =
+    getClients (Proxy :: Proxy sub) (prependSegment schemeAuthority req)
+      where
+        schemeAuthority = printScheme scheme <> printAuthority authority
+
 type RequestBuilder = { path :: Array String }
 
 emptyRequestBuilder :: RequestBuilder
 emptyRequestBuilder = { path: [] }
 
+prependSegment :: String -> RequestBuilder -> RequestBuilder
+prependSegment segment req =
+  req { path = singleton segment <> req.path }
+
 appendSegment :: String -> RequestBuilder -> RequestBuilder
 appendSegment segment req =
   req { path = req.path <> singleton segment }
 
 toAffjaxRequest :: RequestBuilder -> AffjaxRequest Unit
-toAffjaxRequest req = defaultRequest { url = "/" <> joinWith "/" req.path }
+toAffjaxRequest req = defaultRequest { url = joinWith "/" req.path }
```

Comment:
Doesn't this change mean that types without `BaseURI` will be relative? What about a `baseUri :: Maybe String` field in the `RequestBuilder`, or just a `String` defaulting to `/`?

Reply:
Good idea. Adding a field sounds good.